src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/service/SecurityServerServiceTest.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.niis.xroad.centralserver.restapi.domain.ManagementRequestType.CLIENT_REGISTRATION_REQUEST;
+
+@ExtendWith(MockitoExtension.class)
+public class SecurityServerServiceTest implements WithInOrder {
+
+    @Mock
+    private ManagementRequestService managementRequestService;
+
+    @Mock
+    private ManagementRequestInfoDto managementRequestInfoDto;
+
+    @Mock
+    private SecurityServerId serverId;
+
+    @InjectMocks
+    private SecurityServerService securityServerService;
+
+
+    @Nested
+    @DisplayName("findSecurityServerRegistrationStatus(SecurityServerId serverId)")
+    public class SecurityServerRegStatusMethod implements WithInOrder {
+
+        @Test
+        @DisplayName("should find status successfully")
+        public void shouldAddSubsystemSuccessfully() {
+            Page<ManagementRequestInfoDto> requestInfoDtos = new PageImpl<>(List.of(managementRequestInfoDto));
+            doReturn(requestInfoDtos).when(managementRequestService)
+                    .findRequests(ManagementRequestViewRepository.Criteria.builder()
+                            .origin(Origin.SECURITY_SERVER)
+                            .serverId(serverId)
+                            .types(List.of(CLIENT_REGISTRATION_REQUEST))
+                            .build(), Pageable.unpaged());
+            doReturn(ManagementRequestStatus.APPROVED).when(managementRequestInfoDto).getStatus();
+
+            ManagementRequestStatus result = securityServerService.findSecurityServerRegistrationStatus(serverId);
+
+            assertNotNull(result);
+            assertEquals(ManagementRequestStatus.APPROVED, result);
+        }
+
+        @Test
+        @DisplayName("should find status successfully")
+        public void shouldThrowNotFound() {
+            NotFoundException expectedThrows = mock(NotFoundException.class);
+            Page<ManagementRequestInfoDto> requestInfoDtos = Page.empty();
+            doReturn(requestInfoDtos).when(managementRequestService)
+                    .findRequests(ManagementRequestViewRepository.Criteria.builder()
+                            .origin(Origin.SECURITY_SERVER)
+                            .serverId(serverId)
+                            .types(List.of(CLIENT_REGISTRATION_REQUEST))
+                            .build(), Pageable.unpaged());
+
+            Executable testable = () -> securityServerService.findSecurityServerRegistrationStatus(serverId);
+
+            NotFoundException actualThrows = assertThrows(NotFoundException.class, testable);
```

Comment:
the method assertThrows(expectedType, executable) verifies that the executable throws the the exception of expectedType. There is no need to check it again.